gapic/src/main/com/google/gapid/perfetto/views/TraceConfigDialog.java

```diff
@@ -769,16 +769,19 @@ public BasicInputArea(Composite parent, Settings settings, Theme theme,
       addSeparator();
 
       bat = createCheckbox(this, "Battery", sBatt.getEnabled(), e -> updateBat());
-      batLabels = new Label[2];
       Composite batGroup = withLayoutData(
-          createComposite(this, withMargin(new GridLayout(3, false), 5, 0)),
+          createComposite(this, withMargin(new GridLayout(1, false), 5, 0)),
+          withIndents(new GridData(), GROUP_INDENT, 0));
+      batLabels = new Label[2];
+      Composite batLabelGroup = withLayoutData(
+          createComposite(batGroup, withMargin(new GridLayout(3, false), 5, 0)),
           withIndents(new GridData(), GROUP_INDENT, 0));
-      batLabels[0] = createLabel(batGroup, "Poll Rate:");
-      batRate = createSpinner(batGroup, sBatt.getRate(), 250, 60000);
-      batLabels[1] = createLabel(batGroup, "ms");
+      batLabels[0] = createLabel(batLabelGroup, "Poll Rate:");
+      batRate = createSpinner(batLabelGroup, sBatt.getRate(), 250, 60000);
+      batLabels[1] = createLabel(batLabelGroup, "ms");
       if (caps.getHasPowerRail()) {
```

Comment:
It might make more sense to change this so that the checkboxes are always added to the UI, but if power rail is not available:
- we gray out the checkboxes
- and add a label saying something like "This device does not provide information on power consumption and energy breakdown" (feel free to edit to make the flow feel right)

Reply:
`updateBat()` keeps track on these. Majority of OEMs don't provide any information on Power Rails/Energy Breakdown so it was decided not to show those checkboxes instead. (Although the greyed boxes do look better!) I'll discuss on this further and get it in another PR, if necessary.